sklearn/tree/_tree.pyx

```diff
@@ -1142,6 +1277,27 @@ cdef class Tree:
             raise ValueError("Can't initialize array.")
         return arr
 
+    cdef inline double _get_middle_value(self, int node_id) nogil:
+        cdef:
+            int i
+            double middle_value = 0
+
+        if self.max_n_classes == 1:
+            # Regression
+            middle_value = self.value[node_id]
+        else:
+            # Classification
```

Comment:
I thought we forbid multiclass. Binary class could be treated as regression of the positive class. Could you explain your implementation?

Reply:
`max_n_classes` is 2 for binary classification. It's true that this code is quite obscure, let me try to make it more explicit.

Reply:
Done. I think this is clearer now.